include/zephyr/drivers/accel2d.h

```diff
@@ -0,0 +1,680 @@
+ * hardware.
+ * @retval -EIO Underlying 2D accelerator encountered an I/O error.
+ */
+int accel2d_blit_img(const struct device *dev, struct accel_buf *dst_buf,
+		     struct accel_buf *src_buf,
+		     struct accel2d_coords *src_coords,
+		     enum accel2d_rotate rotation,
+		     enum accel2d_scale scale_factor, enum accel2d_blend blend);
+
+/**
+ * @brief Flush a 2D accelerator's render queue
+ *
+ * Flush a 2D accelerator's render queue. Upon return, all buffers
+ * queued for this 2D accelerator in render commands should be updated
+ * with render outputs.
+ * @param dev: 2D accelerator device object
+ * @retval 0 flush completed successfully
+ * @retval -EIO Underlying 2D accelerator encountered an I/O error
+ */
+int accel2d_render(const struct device *dev);
+
+/**
+ * @brief Write directly to a display using the 2D accelerator.
+ *
+ * Write direct to an attached display using the 2D accelerator. This function
+ * will write the provided framebuffer directly to the display, performing pixel
+ * format conversions and transformations automatically.
+ * @param dev: 2D accelerator device object
+ * @param coords: output coordinates to write to on display
+ * @param src_buf: 2D accelerator source buffer.
+ * @param rotation: rotation setting to apply to buffer.
+ * @param scale_factor: scale factor to apply before sending to display
+ * @param output_format: output format for pixels to stream to display.
+ * @retval 0 write completed successfully.
+ * @retval -ENOSYS if this 2D accelerator does not support this feature
+ * @retval -ENOTSUP if the 2D accelerator does not support the
+ *          requested combination of settings
+ * @retval -EINVAL invalid buffer alignment settings for the 2D accelerator
+ * @retval -EBUSY @ref accel2d_render must be called before this operation can
+ *          proceed, as no more rendering can be queued on the 2D accelerator
+ * hardware.
+ * @retval -EIO Underlying 2D accelerator encountered an I/O error.
+ */
+int accel2d_write_display(const struct device *dev,
+			  struct accel2d_coords *coords,
+			  struct accel_buf *src_buf,
+			  enum accel2d_rotate rotation,
+			  enum accel2d_scale scale_factor,
+			  enum display_pixel_format output_format);
+
+/*
+ * @brief Read capabilities of 2D accelerator device
+ *
+ * Read capabilities of 2D accelerator device. This API can be used by graphics
+ * stacks to determine which operations are supported by the 2D accelerator
+ * hardware on their system, so they can fallback to software rendering where
+ * required.
+ * @param dev: 2D accelerator device object
+ * @param caps: 2D accelerator capabilities struct, to be filled by the driver
+ * @retval 0 capabilities were queried successfully
+ * @retval -ENOSYS if the 2D accelerator does not support this API
+ */
+int accel2d_read_caps(const struct device *dev,
+		      struct accel2d_capabilities *caps);
+
+/**
+ * @}
+ */
+
+#ifdef __cplusplus
+}
+#endif
+
```

Comment:
Do we need to consider `CONFIG_USERSPACE`? The `display` API also doesn't use the `__syscall` attributes, just wondering if that is deliberate?